File: aoc2024/day4/problem2.py
from pprint import pprint

input_file = "./input.in"

with open(input_file, "r") as file:
    data = file.read().splitlines()


def findMAS(arr):
    m = len(arr)
    n = len(arr[0])

    count = 0
    valis = {"MS", "SM"}

    for i in range(1, m - 1):
        for j in range(1, n - 1):
            if arr[i][j] != "A":
                continue
            s1 = arr[i - 1][j - 1] + arr[i + 1][j + 1]
            s2 = arr[i - 1][j + 1] + arr[i + 1][j - 1]
            if (s1 in valis) and (s2 in valis):
                count += 1

    return count


print(findMAS(data))

File: aoc2024/day4/problem2.jl
1 + 1

data = readlines("input.in")

count = 0
valid = Set(["MS", "SM"])

for i in 2:(length(data)-1)
    for j in 2:(length(data[1])-1)
        if data[i][j] != 'A'
            continue
        end
        if ((data[i-1][j-1] * data[i+1][j+1]) in valid) && ((data[i-1][j+1] * data[i+1][j-1]) in valid)
            count += 1
        end
    end
end

count